os/tools/mkbinheader.py

```diff
@@ -206,8 +206,12 @@ def get_static_ram_size(bin_type):
         sys.exit(1)
 
     static_ram_size = get_static_ram_size(bin_type)
-    binary_ram_size = int(static_ram_size) + int(dynamic_ram_size)
-    binary_ram_size = roundup_power_two(binary_ram_size)
+    cfg_path = os.getenv('TOPDIR') + '/.config'
+    if check_optimize_config(cfg_path) == True:
+        binary_ram_size = int(dynamic_ram_size)
```

Comment:
For optimization case, static ram size is not needed?
If you think static is also allocated in heap, user cannot think dynamic size should include static size.

Reply:
I understand that you want to give an indication to the user about exact memory requirement for the application. 
For this purpose, i can add some print statement. But, for putting the value in the binary header, I need to put only the dynamic ram. 

In previous scenario, we were allocating one big partition containing text + ro + data + bss + heap. So, I was able to calculate the static and dynamic size here and pass the total size in the binary header. The app would read this size and directly allocate a partition of the size. 

But now, the app allocates memory in 3 parts: round_up(text) , round_up(ro + data) and round_up(data + bss + heap). There is no way for me to pass this information in the header.

Reply:
I think if we need to pass much data, we can expand the header.
What do you think about this?

Reply:
I think the header must be compact. It must contain only the information which the loader cannot get from any other source. Since, the section information can be obtained from the elf, it must not be part of the header. Also, adding section information will make the header tightly coupled with the elf format. If any changes in elf sections or any new format needs to be used, then header will need to be modified. That is why I suggest that the header must contain only the dynamic ram size. If you want to provide an indication to user about the exact ram requirement, it can be given in the form of print logs from the script file or from the runtime logs in the loader which will indicate exact amounts of memory allocated.